### PR TITLE
Update changelog and bump preview version

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: Doxyfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           = "Azure SDK for Embedded C"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = "1.0.0-preview.1"
+PROJECT_NUMBER         = "1.0.0-preview.2"
```
Comment (danewalton) [inline Doxyfile:41]: @danieljurek I just want to make sure this should be updated here. I would assume it would?

Comment (danieljurek) [reply Doxyfile:41]: Your change is correct. Eventually we'll have automation handle pulling that in from `az_version.h`

File: sdk/core/core/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Release History
 
 ## 1.0.0-preview.2 (Unreleased)
+Update top-level CMakeLists.txt to only add subdirectory for specified platform.
+
+Add compilation option to remove all logging from SDK code.
```
Comment (danieljurek) [inline sdk/core/core/CHANGELOG.md:6]:
```suggestion

* Update top-level CMakeLists.txt to only add subdirectory for specified platform.
* Add compilation option to remove all logging from SDK code.
```

Comment (danewalton) [reply sdk/core/core/CHANGELOG.md]: updated

**Pull request merged**